### ISSUE TITLE
[Overlays] You can't see the share icon, when you share a folder/file with other user

### ISSUE THREAD

**Issue by Dianafg76** [owncloud/client#3952]
### Steps to reproduce
1. You need install v 2.0.2 for the desktop, with user4
2. Go to the Server Web with user2
3. Create a new folder (e.g. Share with user4)
4. Click on Share with users, and share with user4
5. Go to the Desktop Client
6. Wait to sync

### Expected behaviour
When the sync is finished, you can see the icon to share 
![screen shot 2015-10-15 at 13 38 04](https://cloud.githubusercontent.com/assets/12032058/10512617/112312c2-7342-11e5-9330-a4d7cbaf5547.png)

### Actual behaviour
When the sync is finished, you can't see the icon to share
![screen shot 2015-10-15 at 13 31 59](https://cloud.githubusercontent.com/assets/12032058/10512509/401636c8-7341-11e5-9a2d-c058d38065ec.png)

NOTE: If you Add an Account, you can se the icon
![screen shot 2015-10-15 at 13 34 51](https://cloud.githubusercontent.com/assets/12032058/10512551/8e83b3d0-7341-11e5-96b4-1aee896e2bfd.png)

### Server configuration
Desktop v ownCloud-2.0.2.2786-nightly20151015.pkg 
Server v {"installed":true,"maintenance":false,"version":"8.2.0.9","versionstring":"8.2 RC1","edition":""}

**Comment by ckamm** [comment]
@Dianafg76 The share icon was displayed directly for me on linux. What do you mean by "If you Add an Account"?

**Comment by Dianafg76** [comment]
@ckamm you need add the new account

**Comment by dragotin** [comment]
@Dianafg76 can you try to explain a bit more elaborated? Where do you need to add the new account? What kind of account? User account? A new account in the client? It's a bit hard to follow.

**Comment by Dianafg76** [comment]
### Steps to reproduce
1. Install th v2.1.0 with 2 user (user1 and user2)
2. Go to the Server Web with user1
3. Create a new folder (e.g. Hello)
4. Share the folder Hello with the user2
5. Go to the Desktop Client with user2
6. Wait to sync


### Actual behaviour
When the sync is finished, you can't see the icon to share. The share icon was not displayed directly for me on MAC

NOTE: If you go to Settings...
2. Click on Geneal Option
3. Click on Add an Account
4. Introduce the URL, and the other user (e.g. user3). You can see the icon

**Comment by jturcotte** [comment]
@Dianafg76 I tried to reproduce it and I also couldn't. I tried with a 8.0 server at first, and then tried with 8.2.2. The way it seems to work is that the server sets a special flag in the WebDAV permission field of each file, the client stores it in its database and is then read to decide if we show a shared icon or not. So it's quite possible that the way you interact with the server is needed to reproduce the issue.

Is there anything other than adding a new account that you think could help reproducing the issue?
I'll change the milestone as the issue is probably not as widespread as initially thought.
